utilities/scripts/xmlPagsToEntityPags.groovy

```diff
@@ -0,0 +1,199 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/**
+ * This file is a groovy script that parses the PAGSGroupStoreConfig.xml and creates entity-PAGS files that
+ * can be imported into an existing uPortal database, creates a SQL script to update the references to the
+ * XML PAGS groups to entity-PAGS groups, and provides some instructions to the user on updating
+ * compositeGroupService.xml to switch from XML PAGS to entity PAGS.
+ *
+ * This script is best IMHO executed directly by groovy as:
+ * groovy xmlPagsToEntityPags.groovy pathToPAGSGroupStoreConfig.xml directoryToWriteEntityFilesTo
+ *
+ * However it can also be invoked via ant as:
+ * ant -Dmaven.test.skip=true -Ddir=directoryToWriteEntityFilesTo data-import
+ * in which case to this groovy script it will look as if invoked as
+ * groovy -s pathToscript xmlPagsToEntityPags.groovy pathToPAGSGroupStoreConfig.xml directoryToWriteEntityFilesTo
+ *
+ * @since 4.2.0
+ */
+
+import groovy.xml.MarkupBuilder
+import org.xml.sax.EntityResolver
+import org.xml.sax.InputSource
+import org.xml.sax.SAXException
+
+// Extra blank output in case script is invoked through ant to segregate it's output better
+println()
+println()
+
+// If groovy script is invoked from ant through uPortal's PortalShell, there will be 2 arguments
+// at the front we can ignore: -s scriptPath
+if (args.length != 2 && args.length != 4) {
+    println "Usage:  groovy xmlPagsToEntityPags.groovy pathToPAGSGroupStoreConfig.xml directoryToWriteEntityFilesTo"
+    println()
+    println "Typically:"
+    println "  pathToPAGSGroupStoreConfig.xml = ../../uportal-war/src/main/resources/properties/groups/PAGSGroupStoreConfig.xml"
+    println "  directoryToWriteEntityFilesTo = ../../uportal-war/src/main/data/myInstitution/pags-group"
+    println "        where myInstitution is your modified copy of the quickstart data set"
+    println()
+    println "Some entities will exist in uportal-war/src/main/data/default_entities/pags-group"
+    println "but those created in your myInstitution will override them"
+    System.exit(1)
+}
+
+def pagsFile = args[args.length == 2 ? 0 : 2]
+def entityDir = args[args.length == 2 ? 1 : 3]
+
+def xmlSlurper = new XmlSlurper()
+
+// Ignore the dtd specified in PAGSGroupStoreConfig.xml.  For some reason I wasn't able to get XmlSlurper
+// to resolve the dtd even when specifying its directory in the classpath.
+xmlSlurper.setEntityResolver(new EntityResolver() {
+    @Override
+    public InputSource resolveEntity(String publicId, String systemId)
+            throws SAXException, IOException {
+        if (systemId.contains("PAGSGroupStore.dtd")) {
+            return new InputSource(new StringReader(""));
+        } else {
+            return null;
+        }
+    }
+})
+// For some reason, must disallow doctype declarations when script is run in intelliJ
+xmlSlurper.setFeature("http://apache.org/xml/features/disallow-doctype-decl", false)
```

Comment:
Running the script in the IDE made debugging MUCH easier, especially for someone who isn't as solid on groovy like me since I could set breakpoints and inspect objects and figure out what they were, etc.  If there are alternative ways to debug using command-line tools I'd love to become more aware of them.